tools/windows-node-installer/test/e2e/azure_test.go

```diff
@@ -0,0 +1,226 @@
+// windows-node-installer.json file specified in "dir".
+func readInstallerInfo() (err error) {
+	wniFilePath := dir + "/windows-node-installer.json"
+	installerInfo, err := resource.ReadInstallerInfo(wniFilePath)
+	if err != nil {
+		return fmt.Errorf("failed to read installer info %s from ARTIFACT_DIR", dir)
+	}
+	if len(installerInfo.SecurityGroupIDs) == 0 {
+		return fmt.Errorf("failed to obtain the sec group Ids")
+	}
+	secGroupsIDs = installerInfo.SecurityGroupIDs
+	if len(installerInfo.InstanceIDs) == 0 {
+		return fmt.Errorf("failed to obtain the instance Ids")
+	}
+	instanceIDs = installerInfo.InstanceIDs
+	return nil
+}
+
+// isWinRMPortOpen returns a bool response if winRmHttps port is opened on
+// windows instance or not.
+func isWinRMPortOpen(secGroupRules []network.SecurityRule) bool {
+	var secRulePropFormat network.SecurityRulePropertiesFormat
+	var destPortRange string
+	var access network.SecurityRuleAccess
+	var protocol network.SecurityRuleProtocol
+	var priority int32
+	var sourceAddressPrefix string
+	for _, secGroupRule := range secGroupRules {
+		if isNil(secGroupRule.SecurityRulePropertiesFormat) {
+			continue
+		}
+		secRulePropFormat = *(secGroupRule.SecurityRulePropertiesFormat)
+		if isNil(secRulePropFormat.DestinationPortRange) || isNil(secRulePropFormat.Priority) ||
+			isNil(secRulePropFormat.SourceAddressPrefix) {
+			continue
+		}
+		destPortRange = *(secRulePropFormat.DestinationPortRange)
+		protocol = secRulePropFormat.Protocol
+		access = secRulePropFormat.Access
+		priority = *(secRulePropFormat.Priority)
+		sourceAddressPrefix = *(secRulePropFormat.SourceAddressPrefix)
+		if destPortRange == winRM && access == winRmAction && protocol == winRMProtocol &&
+			priority == winRMPriority && len(sourceAddressPrefix) != 0 {
+			return true
+		}
+	}
+	return false
+}
+
+// testWinRmPort checks if winRmHttps port is mentioned in the inbound security
+// group rules in the worker subnet.
+func testWinRmPort(t *testing.T) {
+	ctx := context.Background()
+	err := setup()
+	require.NoError(t, err, "failed to initialize azureProvider")
+	err = readInstallerInfo()
+	require.NoError(t, err, "failed to get info from wni file")
+	var secGroupPropFormat network.SecurityGroupPropertiesFormat
+	var secGroupRules []network.SecurityRule
+	for _, nsgName := range secGroupsIDs {
+		secGroupProfile, err := azureInfo.nsgClient.Get(ctx, azureInfo.resourceGroupName, nsgName, "")
+		assert.NoError(t, err, "failed to get the network security group profile")
+		assert.NotNil(t, secGroupProfile.SecurityGroupPropertiesFormat, "failed to get the security group properties format")
+		secGroupPropFormat = *(secGroupProfile.SecurityGroupPropertiesFormat)
+		assert.NotNil(t, secGroupProfile.SecurityRules, "failed to get the security rules list")
+		secGroupRules = *(secGroupPropFormat.SecurityRules)
+		assert.True(t, isWinRMPortOpen(secGroupRules), "winRmHttps port is not opened")
```

Comment:
This should stay an assert, as you are checking the test condition, and the port being false will not cause any issues later in the test - (in this case because this is the end of the test)

Reply:
right, this shouldn't be a require.